test/v1/controllers/SubmitReturnControllerSpec.scala

```diff
@@ -0,0 +1,207 @@
+      |   "vatDueTotal": 	10000,
+      |   "vatReclaimedCurrPeriod": 	1000,
+      |   "vatDueNet": 	9000,
+      |   "totalValueSalesExVAT": 	1000,
+      |   "totalValuePurchasesExVAT": 	200,
+      |   "totalValueGoodsSuppliedExVAT": 	100,
+      |   "totalAllAcquisitionsExVAT": 	540,
+      |   "receivedAt":  "2020-05-05T12:01:00Z",
+      |   "uniqueId": "0123456789"
+      |}
+      |""".stripMargin
+  )
+
+  val submitRequestRawData: SubmitReturnRawData = SubmitReturnRawData(
+    vrn = vrn,
+    body = AnyContentAsJson(submitRequestBodyJson)
+  )
+
+  val requestBody: SubmitReturnRequestBody = SubmitReturnRequestBody(
+    periodKey = "#001",
+    vatDueSales = 7000.00,
+    vatDueAcquisitions = 3000.00,
+    totalVatDue = 10000.00,
+    vatReclaimedCurrPeriod = 1000.00,
+    netVatDue = 9000.00,
+    totalValueSalesExVAT = 1000.00,
+    totalValuePurchasesExVAT = 200.00,
+    totalValueGoodsSuppliedExVAT = 100.00,
+    totalAcquisitionsExVAT = 540.00,
+    receivedAt = "2020-05-05T12:01:00Z",
+    uniqueId = "0123456789"
+  )
+
+  val submitReturnRequest: SubmitReturnRequest =
+    SubmitReturnRequest(
+      vrn = Vrn(vrn),
+      body = requestBody
+    )
+
+  val submitReturnResponseJson: JsValue = Json.parse(
+    """
+      |{
+      |   "processingDate": "2017-01-01T00:00:00.000Z",
+      |   "formBundleNumber": "123456789012",
+      |   "paymentIndicator": "DD",
+      |   "chargeRefNumber": "SKDJGFH9URGT"
+      |}
+      |""".stripMargin
+  )
+
+  "submitReturn" when {
+    "return OK" should {
+      "happy path" in new Test {
+
+        MockSubmitReturnRequestParser
+          .parse(submitRequestRawData)
+          .returns(Right(submitReturnRequest))
+
+        MockSubmitReturnService
+          .submitReturn(submitReturnRequest)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, ()))))
+
+        private val result: Future[Result] = controller.submitReturn(vrn)(fakePostRequest(submitRequestBodyJson))
+
+        status(result) shouldBe CREATED
+        header("X-CorrelationId", result) shouldBe Some(correlationId)
+      }
+    }
+
+    "return the error as per spec" when {
+      "parser errors occur" must {
+        def errorsFromParserTester(error: MtdError, expectedStatus: Int): Unit = {
+          s"a ${error.code} error is returned from the parser" in new Test {
+
+            MockSubmitReturnRequestParser
+              .parse(submitRequestRawData)
+              .returns(Left(ErrorWrapper(correlationId, error, None)))
+
+            val result: Future[Result] = controller.submitReturn(vrn)(fakePostRequest(submitRequestBodyJson))
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (VrnFormatError, BAD_REQUEST),
+          (PeriodKeyFormatError, BAD_REQUEST),
+        )
```

Comment:
All the validation errors should be tested here, it's missing a test for 

FORMAT_VALUE
FORMAT_UNIQUE_ID
FORMAT_RECEIVED_AT